xchange-poloniex2/src/main/java/info/bitrich/xchangestream/poloniex2/dto/PoloniexWebSocketEventsTransaction.java

```diff
@@ -53,31 +50,50 @@ public Long getSeqId() {
-                PoloniexWebSocketOrderbookModifiedEvent insertEvent = new PoloniexWebSocketOrderbookModifiedEvent(orderbookModifiedEvent);
-                events.add(insertEvent);
-            } else if (eventType.equals("t")) {
-                String formatedDate = "";
-                Instant timestamp = Instant.ofEpochSecond(jsonNode.get(5).asInt());
-                SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
-                sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
-                formatedDate = sdf.format(Date.from(timestamp));
-                TradeEvent tradeEvent = new TradeEvent(jsonNode.get(1).asText(),
-                        jsonNode.get(2).asText(),
-                        new BigDecimal(jsonNode.get(3).asText()),
-                        new BigDecimal(jsonNode.get(4).asText()),
-                        formatedDate);
-                PoloniexWebSocketTradeEvent insertEvent = new PoloniexWebSocketTradeEvent(tradeEvent);
-                events.add(insertEvent);
+            switch (eventType) {
+                case "i": {
+                    OrderbookInsertEvent orderbookInsertEvent = mapper.convertValue(jsonNode.get(1), OrderbookInsertEvent.class);
+                    PoloniexWebSocketOrderbookInsertEvent insertEvent = new PoloniexWebSocketOrderbookInsertEvent(orderbookInsertEvent);
+                    events.add(insertEvent);
+                    break;
+                }
+                case "o": {
+                    OrderbookModifiedEvent orderbookModifiedEvent = new OrderbookModifiedEvent(jsonNode.get(1).asText(),
+                            new BigDecimal(jsonNode.get(2).asText()), new BigDecimal(jsonNode.get(3).asText()));
+                    PoloniexWebSocketOrderbookModifiedEvent insertEvent = new PoloniexWebSocketOrderbookModifiedEvent(orderbookModifiedEvent);
+                    events.add(insertEvent);
+                    break;
+                }
+                case "t": {
+                    /*
+                     * ["t", "<trade id>", <1 for buy 0 for sell>, "<price>", "<size>", <timestamp>]
+                     */
+                    String tradeId = jsonNode.get(1).asText();
+                    Order.OrderType side = sideFromString(jsonNode.get(2).asText());
+                    BigDecimal price = new BigDecimal(jsonNode.get(3).asText());
+                    BigDecimal size = new BigDecimal(jsonNode.get(4).asText());
+                    int timestampSeconds = jsonNode.get(5).asInt();
+                    TradeEvent tradeEvent = new TradeEvent(tradeId,
+                            side,
+                            price,
+                            size,
+                            timestampSeconds);
+                    PoloniexWebSocketTradeEvent insertEvent = new PoloniexWebSocketTradeEvent(tradeEvent);
+                    events.add(insertEvent);
+                    break;
+                }
```

Comment:
Needs a `default` case

Reply:
And what to do there? default is noop. we cann't throw an exception there.. if they add new type we should just ignore it

Reply:
On my IDE, and on Sonar analysis, this throws a warning due to the lack of a default condition. It's a code smell. You should be explicit that you are _deliberately_ ignoring the no-op condition:

```
  default: // Ignore
```

This is an [age-old debate](https://stackoverflow.com/questions/4649423/should-switch-statements-always-contain-a-default-clause?lq=1) and I'd rather not open it again. The [Google Style Guide](https://google.github.io/styleguide/javaguide.html#s4.8.4-switch) insists on it, as do most projects I work on, and it's not really much to ask.

Reply:
ok

Reply:
is there a page about contributions? let's add a note about google style guide there. Cause for me it's a question of IDE preferences.  I don't like redundant code, but if it's a local rule that's fine.

Reply:
There isn't a style guide for this repo, but I have been thinking we should add the coveo fmt plugin like XChange and start enforcing Google guidelines, since there is a plan to merge with XChange anyway and it would be good preparatory work.

See long discussion here: https://github.com/knowm/XChange/issues/3220

Reply:
Then we definitely need a separate ticket for this. It's going to be a lot of work anyway. let's keep PRs more atomic:)

Reply:
created a ticket for this https://github.com/bitrich-info/xchange-stream/issues/441

Reply:
Just needs this default case